packages/rs-drive-abci/src/execution/engine/finalize_block_proposal/v0/mod.rs

```diff
@@ -165,25 +181,34 @@ where
             if !validation_result.is_valid() {
                 return Ok(validation_result.into());
             }
+
+            // TODO(withdrawals): We verify withdrawal transactions set what we pass for singing to tenderdash
+            //  is correct on verify_vote_extension. Do we need to verify resulting signatures from tenderdash?
+            //  It does make sense only in case if we don't trust to tenderdash.
+            //  Moreover, if signatures aren't correct then Core RPC will fail when we broadcast these transactions
+            //  to Core. Do we need to verify Tenderdash signatures twice, here in Drive and then in Core?
```

Comment:
Depends on the trust model you take.

Technically, the signature received from Tenderdash should be already verified by Tenderdash, at least with the current code base.

However, I'm a big fan of "zero-trust" approach, which (simplified) says "don't trust other components". In this approach, we should also verify it here.

It should be quite easy, given that rs-tenderdash-abci provides logic for that, as VoteExtension implements [SignDigest](https://github.com/dashpay/rs-tenderdash-abci/blob/556f3a9b6724bdbe44acf8045cd87144a997bccb/abci/src/signatures.rs#L35) trait (still needs to be updated for withdrawals). And there was already some stub impl in Platform, not sure if it's still there.

Reply:
Yes, but we right below broadcast these TXs to Core and it verify signatures and return an error if it's not valid. So we basically using an external service what we trust to verify signatures after tenderdash.

Reply:
Ok, we changed logic, so now we can't rely on Core to make sure that signatures are valid. I'm going to revert this logic back and keep it